docs/community/maintainers-guide/releasing.rst

```diff
@@ -0,0 +1,224 @@
+   ``maint-<major>.<minor>``).
+6. If Travis fails:
+    - Fix issue and ensure head commit is the release commit (i.e. rebase if
+      necessary).
+7. Merge, tag and push release:
+    - ``git merge --ff-only rel-v<major>.<minor>.<patch>``
+    - ``git tag v<major>.<minor>.<patch>``
+    - ``git push inveniosoftware maint-<major>.<minor> v<major>.<minor>.<patch>``
+
+Invenio release
+---------------
+
+The pre-requirement necessary to release Invenio is that all the features in
+the various Invenio modules needed for the release have been merged and
+released. Then:
+
+1. Update the ``setup.py``:
+    - review all modules lower and upper bounds and adjust them as needed.
+2. Documentation:
+    - review documentation and make sure new features or breaking changes are
+      documented, to help users when upgrading.
+3. Prepare release notes (`see example <https://github.com/inveniosoftware/invenio/commit/f4d0aa5ac78d76228fe86754eeb3bbfe81a1854f>`_):
+    - In ``docs/releases/``, copy an existing patch-level or minor
+      release notes (e.g. ``docs/releases/v3.1.2.rst``).
+    - Edit release notes.
+    - Include the new release notes into ``docs/releases/index.rst``.
+    - Check the "Maintenance policy", e.g. is the version correct?
+      (`example <https://github.com/inveniosoftware/invenio/commit/edb863d2f5228fb158c090a69c2db7c3385b6ba3>`_).
+4. Create a release commit (message pattern:
+   ``release: v<major>.<minor>.<patch>``).
+5. Create tag and push release:
+    - ``git tag v<major>.<minor>.<patch>``
+    - ``git push inveniosoftware master v<major>.<minor>.<patch>``
+6. Regularly check ``Travis`` and ``PyPI`` to ensure the release went through.
+
+Manual releases
+---------------
+
+When the process of releasing fails for some reasons, you might want to
+manually publish the new version of a package.
+
+**PyPI**
+
+You can manually reproduce the publishing process done by ``Travis`` by doing:
+
+1. Activate your virtualenv for the package that you want to release.
+2. Generate the different distributions:
+    - ``python setup.py compile_catalog sdist bdist_wheel``
+3. Install the tool to upload releases to PyPI:
+    - ``pip install twine wheel``
+4. Publish:
+    - ``twine upload --repository <repo-name> dist/*``. The command will
+      ask for username and password. Architects should have the credentials.
+
+**NpmJS**
+
+Manual release on ``NpmJS`` is not only needed in case of failure, but also
+when creating a totally new package, never released yet.
+
+1. Make sure that you have an updated version of the
+   `npm <https://www.npmjs.com/get-npm>`_ client installed in your machine.
+2. Login on ``NpmJS``:
+    - ``npm login``: you will need ``inveniosoftware`` username and password.
+      Architects should have the credentials.
+3. Optionally, verify the package before releasing:
+    - ``npm pack``: this will create the final archive that will be
+      published on ``NpmJS`` in case you want to check its content.
```

Comment:
Just for the harmony of the docs: why don't we add this step to the PyPI release? the content of the release can as well be consulted in `sdist` after running `python setup.py sdist`.

Reply:
I am not sure I have understood this comment, sorry :(